src/hotspot/share/gc/shenandoah/shenandoahSATBMarkQueueSet.cpp

```diff
@@ -54,19 +54,7 @@ void ShenandoahSATBMarkQueueSet::filter(SATBMarkQueue& queue) {
   apply_filter(ShenandoahSATBMarkQueueFilterFn(heap), queue);
 }
 
-bool ShenandoahSATBMarkQueueSet::should_enqueue_buffer(SATBMarkQueue& queue) {
-  if (SATBMarkQueueSet::should_enqueue_buffer(queue)) {
-    return true;
-  } else if (queue.index() < buffer_size()) { // Is buffer not empty?
-    Thread* t = Thread::current();
-    if (ShenandoahThreadLocalData::is_force_satb_flush(t)) {
-      // Non-empty buffer is compacted, and we decided not to enqueue it.
-      // We still want to know about leftover work in that buffer eventually.
-      // This avoid dealing with these leftovers during the final-mark, after
-      // the buffers are drained completely. See JDK-8205353 for more discussion.
-      ShenandoahThreadLocalData::set_force_satb_flush(t, false);
-      return true;
-    }
-  }
-  return false;
+void ShenandoahSATBMarkQueueSet::enqueue_completed_buffer(BufferNode* node) {
+  SATBMarkQueueSet::enqueue_completed_buffer(node);
+  Atomic::inc(&_enqueued_count);
```

Comment:
I believe `SATBMarkQueueSet` already tracks this, and we could instead use `SATBMarkQueueSet::completed_buffers_num`?

Reply:
Ohh nice! Will give it a try!